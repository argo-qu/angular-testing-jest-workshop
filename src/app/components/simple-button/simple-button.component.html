<button (click)="buttonClick.emit()">
  {{ text }}
</button>
