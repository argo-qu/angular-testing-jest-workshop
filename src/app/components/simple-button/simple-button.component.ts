import {Component, Input, Output, EventEmitter} from '@angular/core';

@Component({
  selector: 'app-simple-button',
  templateUrl: './simple-button.component.html',
  styleUrls: ['./simple-button.component.scss']
})
export class SimpleButtonComponent {

  @Input() text: string;

  @Output() buttonClick = new EventEmitter<any>();

}
