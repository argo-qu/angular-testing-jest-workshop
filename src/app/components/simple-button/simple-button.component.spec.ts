import { ComponentFixture, TestBed } from '@angular/core/testing';

import { SimpleButtonComponent } from './simple-button.component';

describe('SimpleButtonComponent', () => {
  let component: SimpleButtonComponent;
  let fixture: ComponentFixture<SimpleButtonComponent>;

  beforeEach(async () => {
    await TestBed.configureTestingModule({
      declarations: [ SimpleButtonComponent ]
    }).compileComponents();
  });

  beforeEach(() => {
    fixture = TestBed.createComponent(SimpleButtonComponent);
    component = fixture.componentInstance;
    fixture.detectChanges();
  });

  it('should create', () => {
    expect(component).toBeTruthy();
  });

  it('should look correspondingly to the Input', () => {
    const text = 'Test text 1';
    component.text = text;

    fixture.detectChanges();

    // Способ через fixture
    expect(fixture.debugElement.nativeElement.querySelector('button').innerHTML.trim()).toEqual(text);

    // Способ через snapshot
    // @ts-ignore
    expect(fixture.debugElement.nativeElement.innerHTML).toMatchSnapshot();
  });

  it('should look correctly without input', () => {
    // @ts-ignore
    expect(fixture.debugElement.nativeElement.innerHTML).toMatchSnapshot();
  });

  it('should dispatch event on click', () => {
    spyOn(component.buttonClick, 'emit');

    fixture.debugElement.nativeElement.querySelector('button').click();

    expect(component.buttonClick.emit).toHaveBeenCalled();
  });

});
