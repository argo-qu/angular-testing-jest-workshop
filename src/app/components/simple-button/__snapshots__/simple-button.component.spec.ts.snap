// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SimpleButtonComponent should look correctly without input 1`] = `
"<button> 
</button>"
`;

exports[`SimpleButtonComponent should look correspondingly to the Input 1`] = `
"<button> Test text 1
</button>"
`;
